invenio_pages/services/service.py

```diff
@@ -0,0 +1,35 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2022 CERN.
+#
+# Invenio-Pages is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Page Service API."""
+
+from flask import abort
+from invenio_access.permissions import system_identity
+from invenio_records_resources.services import RecordService
+from sqlalchemy.orm.exc import NoResultFound
+
+from invenio_pages.records.models import PageModel as Page
+
+
+class PageService(RecordService):
+    """Page Service."""
+
+    def read_url(self, url):
+        """Retrieve a page."""
+        return Page.get_by_url(url)
```

Comment:
It's a bit weird that there are two read functions and each returns a different type of data (an object in the first case and serialized data in the second). The reason this was done was because the serialize data of the API is missing one field that the view needs. Maybe using a custom schema for the read_url function would be the way to go but I prefer to check with the reviewers.

Reply:
Also maybe it would make sense to just add the field to the default PageSchema.